Code/tests/test_urltextsearcher.py

```diff
@@ -47,12 +47,14 @@ def setUp(self):
     def tearDown(self):
         pass
 
+    @patch("autopkglib.URLTextSearcher.download_with_curl")
```

Comment:
I don't really like this change, tbh, because I think it makes it slightly more opaque. Each test that uses this function no longer has a choice about what the return values or side effects are when patching the function, because it's done for them at load time of the test case.

For example, if I want to test something more specific with download_with_curl giving me specific output, I can't actually use this function anymore, so I have to copy and paste what it does elsewhere - which essentially eliminates the value of having a primitive function that handles the boilerplate for you.

I'd rather keep this one primitive, and pass in the magic mocks as we currently are.

Reply:
> Each test that uses this function no longer has a choice about what the return values or side effects are when patching the function

Each test have a choice of **not** using this method at all. I suspect for some future test method this will be the case. `match_first` method is there only to have fewer lines of code since most of the code in the test methods is exactly the same.

> For example, if I want to test something more specific with download_with_curl giving me specific output, I can't actually use this function anymore

`match_first` method had already set `mock_download.return_value` before this PR. I don't see how moving the decorator here does change anything. If I apply your logic `match_first` should not manipulate mocked method output in any way -> Test methods should set it instead.

Reply:
In my opinion, this breaks the somewhat independent model of the tests.

Each test should declare its individual mocks, and if it happens to call any subroutines (such as match_first), that value is passed in on call. 

Your change makes it so that these tests don't actually declare their own mocks, which in my opinion makes the purpose of the test slightly less clear. I recognize that it's not a functional change regardless, and that in general Python operation this would be the cleaner approach - but I think unit tests should remain as isolated as possible.

If we someday massively change howe URLTextSearcher works, it will take more work to unpack these unit tests later if they depend on other functions that define their own mocks instead.